include/maths/CClustererStateSerialiser.h

```diff
@@ -43,7 +43,7 @@ struct SDistributionRestoreParams;
 //! to potential competitors.
 class MATHS_EXPORT CClustererStateSerialiser {
 public:
-    using TClusterer1dPtr = std::shared_ptr<CClusterer1d>;
+    using TClusterer1dPtr = std::unique_ptr<CClusterer1d>;
```

Comment:
General problem:

In other parts of the code we use UPtr, so `TClusterer1dPtr` would be `TClusterer1dUPtr`.

Reply:
I thought about this. The problem with renaming the type of pointer is that it makes changes like this much more difficult. Everywhere the typedef is referred to suddenly has to change as well. So this change with become a whole lot bigger. Also we don't use SPtr for shared_ptr for example which would be consistent with this pattern.

Whilst we probably don't need to change this type again, I don't like that an "implementation detail" (type of smart pointer) gets mixed up with the name of the type. In some cases we can't avoid this because we have a `shared_ptr<T>` and a `unique_ptr<T>` in the same scope but in most cases we can. 

Thoughts?

Reply:
I am fine with that, actually I agree more to what you wrote, for me one purpose of a typedef/using is to abstract away the implementation detail, putting it back in the name defeats this. 

I just wanted to point it out, if you grep the codebase for `UPtr` you find a small number of uses but I could not find a rule in our new and old styleguide.

Any other comments on this?

Reply:
Yes I think our naming of pointer types is not completely consistent. I would propose something like the following rules:
1. If there is only one type of smart pointer in scope then it should be suffixed with Ptr
2. If there would be a name collision then they should be all qualified: shared_ptr as SPtr, unique_ptr as UPtr, auto_ptr as APtr and weak_ptr as WPtr.

Let's take this discussion offline though. If we get agreement on this we can revisit the codebase and tidy up the naming to conform to this in a separate PR.